### PR TITLE
Connect

### PR DESCRIPTION
**Pull request by abarger**
Updated with the MIDHT upgrades to the CONNECTUniversalClientGUI.

### CODE REVIEW

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml [hunk 1]
```diff
@@ -31,6 +31,10 @@
   </configurations>
 
   <dependencies>
+    <!-- To find dependency params use http://www.mavensearch.net/ to find the XML descriptor.
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml:34]: This comment should be removed. mavensearch.net, as far as I can tell, is a spam website that is not related to maven at all. Other organizations have removed links to mavensearch.net, citing them as nsfw for their organizations.

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/AdapterCommonDataLayer.java [hunk 2]
```diff
@@ -1,29 +1,34 @@
 /*
- * Copyright (c) 2012, United States Government, as represented by the Secretary of Health and Human Services. 
- * All rights reserved. 
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *  
+ * Copyright 2010(Year date of delivery) United States Government, as represented by the Secretary of Health and Human Services.  All rights reserved.
+ *  
+ */
+/*
+ * Copyright (c) 2011, Conemaugh Valley Memorial Hospital
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/AdapterCommonDataLayer.java]: Conemaugh's copyright info has overwritten CONNECT's.

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml (hunk 1 above)
Comment (hoffmabc) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml:34]: How does that comment make it into the build?

Thanks,

Brian Hoffman
Lead Associate
Booz | Allen | Hamilton
Supporting DoD P&R IM / MHS

On Aug 30, 2012, at 10:49 AM, "zmelnick" <notifications@github.com<mailto:notifications@github.com>> wrote:

In Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml:

> @@ -31,6 +31,10 @@
>    </configurations>
> 
>    <dependencies>
> -    <!-- To find dependency params use http://www.mavensearch.net/ to find the XML descriptor.

This comment should be removed. mavensearch.nethttp://mavensearch.net, as far as I can tell, is a spam website that is not related to maven at all. Other organizations have removed links to mavensearch.nethttp://mavensearch.net, citing them as nsfw for their organizations.

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/catalog.xml [hunk 3]
```diff
@@ -169,97 +169,97 @@
...
-    <system systemId="file:/C:/Projects/NHINC/Current/Product/Production/Common/Interfaces/src/schemas/HL7V3/NE2008/multicacheschemas/COCT_MT260003UV.xsd" uri="src/main/resources/xml-resources/web-service-references/DODConnector/schemas/HL7V3/NE2008/multicacheschemas/COCT_MT260003UV.xsd"/>
-    <system systemId="file:/C:/Projects/NHINC/Current/Product/Production/Common/Interfaces/src/schemas/HL7V3/NE2008/multicacheschemas/COCT_MT240000UV01.xsd" uri="src/main/resources/xml-resources/web-service-references/DODConnector/schemas/HL7V3/NE2008/multicacheschemas/COCT_MT240000UV01.xsd"/>
-    <system systemId="file:/C:/Projects/NHINC/Current/Product/Production/Common/Interfaces/src/schemas/HL7V3/NE2008/multicacheschemas/COCT_MT140007UV.xsd" uri="src/main/resources/xml-resources/web-service-references/DODConnector/schemas/HL7V3/NE2008/multicacheschemas/COCT_MT140007UV.xsd"/>
+    <system systemId="file:/C:/Projects/NHINC/3.1/Product/Production/Common/Interfaces/src/wsdl/DODConnector.wsdl" uri="src/main/resources/xml-resources/web-service-references/DODConnector/wsdl/DODConnector.wsdl"/>
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/catalog.xml]: If this is for 3.3, then the 3.1 should be changed to 3.3

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml (hunk 1 above)
Comment (bhumphrey) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/ivy.xml:34]: This was removed by a previous pull request and shouldn't be included here.

**Comment by bhumphrey** [comment]
This pull request needs be for 3.3_integration not CONNECT_integration. CONNECT_integration is the next release line (4.0).

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/AdapterCommonDataLayer.java [hunk 4]
```diff
@@ -31,108 +36,130 @@
...
 @WebService(serviceName = "CommonDataLayerService", portName = "CommonDataLayerPort", endpointInterface = "gov.hhs.fha.nhinc.adapter.commondatalayer.CommonDataLayerPortType", targetNamespace = "urn:gov:hhs:fha:nhinc:adapter:commondatalayer", wsdlLocation = "META-INF/wsdl/AdapterCommonDataLayer/AdapterCommonDataLayer.wsdl")
 @BindingType(value = javax.xml.ws.soap.SOAPBinding.SOAP12HTTP_BINDING)
 @Stateless
-public class AdapterCommonDataLayer implements CommonDataLayerPortType {
+public class AdapterCommonDataLayer implements CommonDataLayerPortType
+{
+
+    @Override
+    public org.hl7.v3.FindDocumentRCMRIN000032UV01ResponseType findDocument(org.hl7.v3.FindDocumentRCMRIN000031UV01RequestType param0)
+    {
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/AdapterCommonDataLayer.java]: Lines here have been made longer than 120 characters. Consider using the code formatting guidelines, documented here: https://developer.connectopensource.org/display/CONNECTWIKI/Coding+templates+and+formatters

CI checks for formatting violations.

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/CommonDataLayerService.java [hunk 5]
```diff
@@ -1,146 +1,100 @@
...
- * 
  * JAX-WS RI 2.1.3.1-hudson-749-SNAPSHOT
- * 
  * Generated source version: 2.1
  * 
- * 
  */
-
-@WebServiceClient(name = "CommonDataLayerService", targetNamespace = "urn:gov:hhs:fha:nhinc:adapter:commondatalayer", wsdlLocation = "file:/D:/GlassFishESBV2.1/NetBeansProjects6.5.1/NHINC/Current/Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/conf/xml-resources/web-services/AdapterCommonDataLayer/wsdl/AdapterCommonDataLayer.wsdl")
+@WebServiceClient(name = "CommonDataLayerService", targetNamespace = "urn:gov:hhs:fha:nhinc:adapter:commondatalayer", wsdlLocation = "file:/D:/GlassFishESBV2.1/NetBeansProjects6.5.1/NHINC/3.1/Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/conf/xml-resources/web-services/AdapterCommonDataLayer/wsdl/AdapterCommonDataLayer.wsdl")
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/CommonDataLayerService.java]: Is 3.1 correct as a directory? Is this for 3.3?

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/mappers/JAXBUtil.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ *
+ * @author A22387
+ */
+public class JAXBUtil
+{
+
+    private static JAXBUtil instance = null;
+    private static JAXBContext jcOrgHl7V3 = null;
+    private static JAXBContext jcIHE = null;
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/mappers/JAXBUtil.java]: This field does not appear to be used. Dead code.

**Comment by abarger** [comment]
Sorry about submitting the pull request on the wrong branch. I'll make sure the next submission is on the integration line.

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/mappers/StaticUtil.java [hunk 7]
```diff
@@ -1,92 +1,450 @@
...
+        JAXBElement element = null;
+
+        try
+        {
+            //read file in line by line using a String Builder to build the xml doc from the FileInputStream
+            FileInputStream fis = new FileInputStream(dataFile);
+            BufferedReader br = new BufferedReader(new InputStreamReader(fis));
+            StringBuilder sb = new StringBuilder();
+
+            for(;;)
```
Comment (zmelnick) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/mappers/StaticUtil.java]: consider using while(true). While they both do the same thing, while(true) is more readily understood. No change really necessary though.

File: Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/CommonDataLayerService.java (hunk 5 above)
Comment (abarger) [inline Product/Production/Adapters/Framework/AdapterCommonDataLayerEJB/src/main/java/gov/hhs/fha/nhinc/adapter/commondatalayer/CommonDataLayerService.java]: 3.1 is not correct. The AdapterCommonDataLayer code (and all of the other projects in Adapter\Framework directory) is being updated for use with CONNECT v3.3.1.  I will make sure to incorporate this change and the other suggested changes into the updated version.

**Comment by abarger** [comment]
I'll make sure the Doc Assembly Plug-in code has the correct Goverment copyright header. Please let me know if copyright changes need to be made to the CONNECTUniversalClientGUI code. Thanks.

**Pull request closed without merging**